tools/distpartitioning/initialize.py

```diff
@@ -0,0 +1,474 @@
+    This is used by non-rank-0 processes. 
+
+    Parameters:
+    -----------
+    rank : integer
+        rank of the process
+    world_size : integer
+        no. of processes used
+
+    Returns:
+    --------
+    numpy ndarray
+        node_feature data, of floats, as received by the scatter_object_list function
+    """
+    send_objs = [None for _ in range(world_size)]
+    recv_obj = [None]
+    dist.scatter_object_list(recv_obj, send_objs, src=0)
+
+    node_features = recv_obj[0]
+    return node_features
+
+def read_graph_files(rank, params, node_part_ids): 
+    """
+    Read the files and return the data structures
+    Node data as read from files, which is in the following format: 
+        <node_type> <weight1> <weight2> <weight3> <weight4> <global_type_nid> <attributes>
+    is converted to 
+        <node_type> <weight1> <weight2> <weight3> <weight4> <global_type_nid> <nid> <recv_proc>
+    Edge data as read from files, which is in the following format: 
+        <global_src_id> <global_dst_id> <global_type_eid> <edge_type> <attributes>
+    is converted to the following format in this function:
+        <global_src_id> <global_dst_id> <global_type_eid> <edge_type> <recv_proc>
+
+    Parameters:
+    -----------
+    rank : integer
+        rank of the process
+    params : argparser object
+        argument parser data structure to access command line arguments
+    node_part_ids : numpy array 
+        array of part_ids indexed by global_nid
+
+    Returns:
+    --------
+    numpy ndarray
+        integer node_data with additional columns added
+    numpy ndarray
+        floats, node_features as read from the node features file
+    numpy ndarray
+        integer, edge_data with additional columns added
+    numpy ndarray
+        floats, edge_features are read from the edge feature file
+    """
+    augmted_node_data = []
+    node_data = read_nodes_file(params.input_dir+'/'+params.nodes_file)
+    augmted_node_data = augment_node_data(node_data, node_part_ids)
+    print('Rank: ', rank, ', Completed loading nodes data: ', augmted_node_data.shape)
+
+    augmted_edge_data = []
+    edge_data = read_edges_file(params.input_dir+'/'+params.edges_file)
+    removed_edge_data = read_edges_file(params.input_dir+'/'+params.removed_edges)
+    edge_data = np.vstack((edge_data, removed_edge_data))
+    augmted_edge_data = augment_edge_data(edge_data, node_part_ids)
+    print('Rank: ', rank, ', Completed loading edges data: ', augmted_edge_data.shape)
+
+    node_features = {}
+    node_features = read_node_features_file( params.input_dir+'/'+params.node_feats_file )
+    print('Rank: ', rank, ', Completed loading node features reading from file ', len(node_features))
+
+    edge_features = {}
+    #edge_features = read_edge_features_file( params.input_dir+'/'+params.edge_feats_file )
+    #print( 'Rank: ', rank, ', Completed edge features reading from file ', len(edge_features) )
```

Comment:
remove

Reply:
I want to leave this commented line in the code as it is. I want to use this as a place-holder for the test cases which has edge features in the next deliverables. Also, I do not want to change the return types of functions now, otherwise I will end up testing this change and it will take a few more hours to complete this one code review. I will address this in the next deliverable, if this is indeed needed.